relay-server/src/services/processor/span/processing.rs

```diff
@@ -59,29 +59,32 @@ enum ValidationError {
     MissingExclusiveTime,
 }
 
-#[allow(clippy::too_many_arguments)]
 pub async fn process(
     managed_envelope: &mut TypedEnvelope<SpanGroup>,
     event: &mut Annotated<Event>,
     extracted_metrics: &mut ProcessingExtractedMetrics,
     project_id: ProjectId,
     ctx: processing::Context<'_>,
     geo_lookup: &GeoIpLookup,
-    reservoir_counters: &ReservoirEvaluator<'_>,
 ) {
     use relay_event_normalization::RemoveOtherProcessor;
 
-    // We only implement trace-based sampling rules for now, which can be computed
-    // once for all spans in the envelope.
-    let sampling_result = dynamic_sampling::run(
-        managed_envelope,
-        event,
-        ctx.config,
-        ctx.project_info,
-        ctx.sampling_project_info,
-        reservoir_counters,
-    )
-    .await;
+    // If no metrics could be extracted, do not sample anything.
+    let should_sample = matches!(&ctx.project_info.config().metric_extraction, ErrorBoundary::Ok(c) if c.is_supported());
```

Comment:
This check was previously evaluated in `supports_sampling` in the `Sampling` trait.